File: src_code/post_proc/read_mslice.m
%Read mass average output from simulation
%clear all

%Store Present Working directory
pwdir = pwd;
if (exist('resultfile_dir') == 0)
    resultfile_dir = './../results';
    display('setting results file to default "./../results"');
end

%Read simulation properties from header file and calculate simulation
%properties
read_header;
Nmass_records = floor(Nsteps / (tplot * Nmass_ave));

%Load mass CV data
cd(resultfile_dir);
fid = fopen('./mslice','r','ieee-le');
cd (pwdir);
%Check file exists
if (fid == -1)
    error('mslice file does not exist in results')
end
mbins = fread(fid,'int32');
mass_slice = reshape(mbins,gnbins(mass_outflag),Nmass_records);
fclose(fid);

%Get two directions orthogonal to slice direction
jxyz = mod(mass_outflag+1,3)+1;
kxyz = mod(mass_outflag,3)+1;

density_slice = mass_slice/(Nvel_ave*nbins(jxyz)*nbins(kxyz)*binsize(1)*binsize(2)*binsize(3));

%Plot evolution
% for i = 1:Nmass_records
% %   plot(mass_slice(:,i));
%    plot(density_slice(:,i));
%    pause(0.5);
% end

File: src_code/post_proc/python/read_mslice.py
#! /usr/bin/env python
###############################################################################
# read_mslice.py
# author: David Trevelyan
#
# read_mslice.py reads the binary data from the file 'mslice' and stores it in 
# a numpy array. This can then be used as part of read_vslice to plot the 
# average velocity profile from a simulation.
# 
# -----------------------------------------------------------------------------
from numpy import *
from array import array
import read_header
import math
import os

# Retrieve important variables from simulation_header with read_header.py
Nsteps        = int(read_header.Nsteps)
tplot         = int(read_header.tplot)
Nmass_ave     = int(read_header.Nmass_ave)
initialstep   = int(read_header.initialstep)
m_outflag     = int(read_header.mass_outflag)
gnbins        =[int(read_header.gnbins1),
                int(read_header.gnbins2),
                int(read_header.gnbins3)]
Nmass_records = int(math.floor((Nsteps-initialstep)/(tplot*Nmass_ave)))
nbins         = int(gnbins[m_outflag-1])

f = open('../results/mslice','rb')             # Create binary file object
mslice = array('i')                            # Initialise array of integers
mslice.fromfile(f,nbins*Nmass_records)         # Read binary file into array
mslice = reshape(mslice,(Nmass_records,nbins)) # Reshape array into 3D
f.close()                                      # Close file object
